### ISSUE TITLE
Release 7.1.0 breaks arrays in documents - each element is nested in an array

### ISSUE THREAD

**Issue by richarddd** [typegoose/typegoose#300]
## Versions
- NodeJS: 12.0.0
- Typegoose(NPM): 7.x.x
- Typegoose(GIT): commithash
- mongoose: 5.9.20
- mongodb: 3.5.9

## What is the Problem?

Arrays prior to 7.1.0 are stored correctly. In versions later than 7.1 each element is nested in an own array:
## Code Example

```ts
class Example {
  @prop()
  public arrayExample!: string[]
}

const arrayExample: ["1","2","3"]

new Example({arrayExample}).save()
```

My expectations in DB are:
```javascript
{
   _id: ...,
   arrayExample: ["1","2","3"]
}
```

Its instead:
```javascript
{
   _id: ...,
   arrayExample: [["1"],["2"],["3"]]
}
```

## Do you know *why* it happenes?

*no*

**Comment by smolinari** [comment]
```
class Something {
  @prop({ items: String })
  public languages?: string[];
}
```
Note the `items:String`. 

Scott

**Comment by hasezoey** [comment]
as @smolinari already said, `items: the.actual.type` (or now `type: the.actual.type`) is still needed, it was already needed prior to 7.1

**Comment by hasezoey** [comment]
updated title to better reflect what this issue is (because of duplicates)

**Comment by henhen724** [comment]
The "items" property is now deprecated what would be the correct way to fix this now?  When I set type to [String] I get Error: "Example.arrayExample"'s Type is invalid! Type is: "function String() { [native code] }" [E009].  I am only able to fix this by setting type to Schema.Types.Mixed.

**Comment by hasezoey** [comment]
@henhen724 please give an example on how you try to set it
(btw, did you try to do `type: [String]`? when yes, this is only available since 7.4.0)

**Comment by henhen724** [comment]
Yes I did use type: [String], but I am using 7.3.5 so it should work.  Here is a repo with a minimal reproducible example: https://github.com/henhen724/typegoose-issue-example.  Both the dev and start will throw the error I'm talking about.

**Comment by hasezoey** [comment]
like i said in my earlier comment

> did you try to do type: [String]? when yes, this is only available since 7.4.0

this syntax is only available since 7.4.0

**Comment by henhen724** [comment]
Sorry, I missed read that

**Comment by SaulMoreyra** [comment]
> @henhen724 please give an example on how you try to set it (btw, did you try to do `type: [String]`? when yes, this is only available since 7.4.0)

````
@ObjectType("location")
export class LocationInformation {
  @Field(() => String, { description: "User location latitude" })
  @prop({ required: true })
  public type: string;

  @Field(() => [Number], { description: "User location longitude" })
  @prop({ required: true, type: [Number] })
  public coordinates: number[];
}
```

**Comment by hasezoey** [comment]
@SaulMoreyra if you have a issue, please open a new issue, because this one is quite old with outdated information
